### ISSUE TITLE
Cloud.typography Uncaught (in promise)

### ISSUE THREAD

**Issue by yumyo** [bramstein/fontfaceobserver#24]
Using cloud.typography with production settings (fonts are served from local server filesystem) I can't get around an Uncaught (in promise) error.

**Comment by yumyo** [comment]
strange enough it works perfecty coping the lib from Filament [https://github.com/filamentgroup/font-loading/blob/master/font-events.html](https://github.com/filamentgroup/font-loading/blob/master/font-events.html)

**Comment by bramstein** [comment]
Do you have an example page somewhere?

**Comment by bpartridge83** [comment]
This _could_ be due to cloud.typography splitting the @font-face definitions into A and B.  For example, we're using "Cyclone Inline", but the @font-faces available are "Cyclone Inline A" and "Cyclone Inline B".

I had a really long explanation of using custom test strings based on the available characters in each defintion (ie. "BMSbswy" vs "BESbswy", because of the missing "E" in the A version), but it turns out you can check both versions simultaneously with a single observer.

Fails: `new FontFaceObserver('Cyclone Inline', ...`, `.check().then(...`
Fails: `new FontFaceObserver('Cyclone Inline A', ...`, `.check().then(...`
Succeeds: `new FontFaceObserver('Cyclone Inline A', ...`, `.check('BMSbswy').then(...`
Succeeds: `new FontFaceObserver('Cyclone Inline A, Cyclone Inline B', ...`, `.check().then(...`

The last example is probably the best option.

**Comment by iamcarrico** [comment]
We had the same issue a while back, @jonsuh discovered it was because of the spaces within the font names for typography.com, "Whitney A" did not work, but "Whitney%20A" did. I have not tested full browser support for what uses "A" vs "B" for typography.com fonts yet.

**Comment by bramstein** [comment]
@iamcarrico That sounds odd. Does typography.com generate @font-face rules with `%20` for spaces?

**Comment by di5abled** [comment]
I had the same issue with local hosted fonts from fonts.com

I solved the issue by encoding the font names in the observer eg: `Trebuchet%20W01%20Regular` instead of  `Trebuchet W01 Regular`.

Also the names of the font faced defined in the css don't always match the "real" name of the font. I found the name which I had to use in the .svg version of the downloaded fonts. `<font-face font-family="Trebuchet W01 Regular" ...`
